### ISSUE TITLE
InContextLearningGenerationExactMatchAccuracy: a gold alias that normalizes to empty (bare article, punctuation-only) makes any reply score correct

### ISSUE THREAD

**Issue by shaurya416** [mosaicml/llm-foundry#1897]
`llmfoundry/eval/metrics/nlp.py:136-137` (`normalize_answer`/`remove_articles`) and `llmfoundry/eval/metrics/nlp.py:198-205` (`InContextLearningGenerationExactMatchAccuracy.update`, scoring branch):

```python
def remove_articles(text: str) -> str:
    return re.sub(r'\b(a|an|the)\b', ' ', text)
...
            if any(
                cleaned_final_answer.startswith(label)
                for label in cleaned_sample_labels
            ):
                self.correct += torch.tensor(1.0)
                metric_result_dict['result'].append(1)
            else:
                metric_result_dict['result'].append(0)
```

`normalize_answer()` strips articles, punctuation and whitespace before comparing. If a gold answer/alias is itself a bare article or made up entirely of punctuation (e.g. `answer = "The"`), it normalizes to `''`. The match is `cleaned_final_answer.startswith(label)`, and every string in Python starts with `''`, so the example scores correct regardless of the model's output — including empty output, whitespace-only output, garbage, and an explicit refusal.

## What happens (reproduced on the real module)

Confirmed as reported. InContextLearningGenerationExactMatchAccuracy.update() (llmfoundry/eval/metrics/nlp.py:198-205) uses cleaned_final_answer.startswith(label) with no guard against a normalized label being the empty string; normalize_answer's remove_articles (nlp.py:136-137) plus punctuation stripping means a gold alias like 'The', 'A', or a punctuation-only string ('...') normalizes to '', which every candidate string trivially starts with, so the example scores correct regardless of model output. Gold aliases come unfiltered from the dataset's 'answer'/'aliases' JSONL fields with no upstream sanitization, so any ICL generation-QA dataset with a bare-article/stopword or punctuation-only answer/alias silently inflates the reported accuracy with no warning. Fix: drop labels that normalize_answer() reduces to '' from cleaned_sample_labels before the startswith check (or fall back to exact equality when normalization empties a label), mirroring the existing non-empty guards used for cot_delimiter/stopping_criteria in the same method.

Happy to open the PR.
